### ISSUE TITLE
Eosjs with scatter - couldn't make use of signature from scatter while performing transactions

### ISSUE THREAD

**Issue by amolmaid** [EOSIO/eosjs#293]
I am trying to use _eosjs_ along with _scatter_ in my **web Application with ReactJS**, but I couldn't figure out how to use _scatter signature_ in place of eosjs permission while performing transactions.

**1. I tried following code snippet, but its giving me error** 
![code](https://user-images.githubusercontent.com/7837458/43822158-ead9da64-9b08-11e8-92c7-e7f08f4d8461.png)

But its giving me error,
`Failed for eosScatter transfer  TypeError: t(...)[u] is not a function
    at Promise (inject.js:40)
    at new Promise (<anonymous>)
    at Proxy.n (inject.js:40)
    at CreateAccount.js:139`

2. I also used following code to bring the get permission for window, but still not working 
![error code 2](https://user-images.githubusercontent.com/7837458/43834883-7bcdb71a-9b2d-11e8-836e-dedb46438b40.png)

and giving following error
![error account missing](https://user-images.githubusercontent.com/7837458/43834893-8468aa74-9b2d-11e8-82a5-5fca02470683.png)

**_could anyone guide me how to use scatter signature in eosjs apis with proper syntax, OR how to pop this get signature window and then perform transaction_**

**3. Also I am getting error**(TypeError: Cannot read property 'find' of undefined) **while getting   identity.accounts**
`console.log("identity = ", scatter.identity); // it gives identity with {hash, key, name, publicKey} object`
`const account = scatter.identity.accounts.find(account => account.blockchain === 'eos');`

Please guide for above

**Comment by chris-allnutt** [comment]
@nsjames may be able to help.  I'm unfamiliar with scatter's signing but thought it handles sending the transaction for you as well.

**Comment by amolmaid** [comment]
It would be really nice if he would! As I simply need to know _how exactly I can get the permission from scatter to perform actions which needs authorization(like transfer or set contract)_; I need **exact syntax** as documentation is not in depth for me.

Thanks in advance.

**Comment by amolmaid** [comment]
Hi,
   Actually I was able to find the solution with the help of scatter telegram group.
1. Basically answer to my 1st and 2nd query was, I should pass **account name on which I have published a contract** rather than contract name in method `eosScatter.contract()`
![solution](https://user-images.githubusercontent.com/7837458/43942686-9ad0c836-9c96-11e8-91bc-caab493eeafb.png)

2. And answer to the third query is
Scatter identity was not picking properly with actual account info, so to do that we need to create scatter object using network object along with the chainId added explicitly.
-- Also don't forget to add chain id in network field of scatter plugin.

anyway Thanks though :)
